src/k8s/clients.py

```diff
@@ -0,0 +1,154 @@
+"""Different implementations of k8s clients."""
+from copy import deepcopy
+from threading import Lock
```

Comment:
does threading.lock work since there might be multiple worker processes on sanic?

Reply:
That is a good point. I am not sure but just to be safe I switched it to multiprocessing.Lock. It is awesome that both support context managers. Also this did not matter all that much because it is only used in the dummy adapters. But still it is better to be safe than get stuck chasing weird errors because of some bug like this.